src/dom.js

```diff
@@ -29,6 +30,20 @@ const HTML_ESCAPE_REGEX = /(&|<|>|"|'|`)/g;
 
 
 /**
+ * Determines if this element is an AMP element
+ * @param {!Element}
+ * @return {boolean}
+ */
+export function isAmpElement(element) {
+  const tag = element.tagName;
+  // Use prefix to recognize AMP element. This is necessary because stub
+  // may not be attached yet.
+  return startsWith(tag, 'AMP-') &&
```

Comment:
From the [original comment](https://github.com/ampproject/amphtml/pull/9279/files#diff-9379e12fba31cf69d032da3c00ba0c8eL372)

> We can tell whether the parent has been stubbed by whether a resource has been attached to it.

That means we have to be able to tell regardless of where the element is in the CustomElement lifecycle. And an element has to have been stubbed (`#connectedCallback`, the first step) to have the `i-amphtml-element` class.